src/features/posts/posts.ts

```diff
@@ -0,0 +1,54 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type InitialState = {
+  loaded: boolean;
+  hasError: boolean;
+  posts: Post[];
+  selectedPost: Post | null;
+};
+
+const initialState: InitialState = {
+  loaded: false,
+  hasError: false,
+  posts: [],
+  selectedPost: null,
+};
+
+export const init = createAsyncThunk(
+  'posts/fetch',
+  (userId: number | object) => {
+    if (typeof userId === 'number') {
+      return getUserPosts(userId);
+    } else {
+      return [];
+    }
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setSelectedPost: (state, action) => {
+      return { ...state, selectedPost: action.payload };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(init.pending, state => {
+      return { ...state, loaded: false };
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      return { ...state, posts: action.payload, loaded: true };
+    });
+
+    builder.addCase(init.rejected, state => {
+      return { ...state, hasError: true, loaded: false };
```

Comment:
In the `init.rejected` case, consider setting `loaded` to `true` to indicate that the loading process has completed, even though it resulted in an error. This will align with the typical pattern of setting `loaded` to `true` upon completion of any async operation.